cluster-autoscaler/debuggingsnapshot/debugging_snapshotter.go

```diff
@@ -0,0 +1,222 @@
+/*
```

Comment:
This file the core part of this PR. Please add some unit tests to confirm it is working as intended.